networkx/utils/tests/test_random_sequence.py

```diff
@@ -47,3 +48,11 @@ def test_random_weighted_choice():
     c = nx.utils.weighted_choice(mapping, seed=1)
     c = nx.utils.weighted_choice(mapping)
     assert c == "a"
+
+
+def test_random_sequence_low_precision():
+    rng = np.random.default_rng(seed=17)
+    x = np.exp(rng.normal(0, 1, 1000)).astype(np.float16)
+    cdf = nx.utils.cumulative_distribution(x)
+    assert len(cdf) == 1001
+    assert cdf[-1] == 1.0
```

Comment:
Unit test is failing because it depends on `numpy`. Here is a version that doesn't:

```suggestion
    nx.utils.cumulative_distribution([0.1] * 100)[-1] == 1.0
```